File: 1759_Count_Number_of_Homogenous_Substrings/count.cpp
class Solution {
public:
    int countHomogenous(string s) {
        long long count = 0;
        long long res = 0;
        char prev;
        int m = 1000000007;
        for(auto i:s){
            if(i==prev){
                count++;
            }
            else{
                count=1;
            }
            res=res+count;
            prev = i;
        }
        return res%m;
    }
};

File: 1759_Count_Number_of_Homogenous_Substrings/count.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        count = 0
        prev = ''
        res = 0
        m = 1000000007

        for i in s:
            if i == prev:
                count+=1
            else:
                count = 1
            res+=count
            prev = i 
        return res%m